internal/apiserver/server.go

```diff
@@ -293,7 +293,19 @@ func (as *apiServer) apiWrapper(handler func(res http.ResponseWriter, req *http.
 		// and the caller can either listen on the websocket for updates, or poll the status of the affected object.
 		// This is dependent on the context being passed down through to all blocking operations down the stack
 		// (while avoiding passing the context to asynchronous tasks that are dispatched as a result of the request)
-		ctx, cancel := context.WithTimeout(req.Context(), as.apiTimeout)
+		reqTimeout := as.apiTimeout
+		reqTimeoutHeader := req.Header.Get("Request-Timeout")
+		if reqTimeoutHeader != "" {
+			customTimeout, err := fftypes.ParseDurationString(reqTimeoutHeader, time.Second /* default is seconds */)
+			if err != nil {
+				log.L(req.Context()).Warnf("Invalid Request-Timeout header '%s': %s", reqTimeoutHeader, err)
+			} else {
+				reqTimeout = time.Duration(customTimeout)
+			}
+		}
```

Comment:
Do we want to have a `maxRequestTimeout` to have an upper bound on this value? If someone sets an absurdly large `Request-Timeout` maybe we should return a `204`? If FireFly is deployed behind a load balancer or reverse proxy, admins may want to configure a `maxRequestTimeout` to not exceed the timeout of the network device in front of FireFly. Otherwise requests could get timeouts from the load balancer/proxy which would look different than timeouts from FireFly itself.

Reply:
Sounds great 👍  - will add that